plugins/org.jboss.tools.openshift.io.core/plugin.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?eclipse version="3.5"?>
+<plugin>
+   <extension-point id="org.jboss.tools.openshift.io.core.tokenProvider" name="tokenProvider" schema="schema/org.jboss.tools.openshift.io.core.tokenProvider.exsd"/>
```

Comment:
because I want to have loose coupling so that extension user can cast to Function<IResource, String>

Reply:
how would both of these not be possible with an osgi service?

Reply:
I thought registering/retrieving an OSGI service was only possible through a Class object but it would be possible with any string. Will try if I can before the code freeze but the LoginProvider would stay as is because I planned to used priority based selection